### PR TITLE
Bundle localkube in the minikube binary as a blob, send that to the VM.

### PR DESCRIPTION
**Pull request by dlorenc**
This PR changes the Makefile a bit to do proper dependency management.

It now builds localkube (via docker on Mac) before the minikube build, then packs it using go-bindata. Minikube uses native scp to transfer this file into the VM, and starts that binary instead of downloading it over the internet.

cc @luxas

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,5 @@ _testmain.go
 
 /out
 /.gopath
+
+pkg/minikube/cluster/localkubecontents.go
```
Comment (luxas) [inline .gitignore]: nit: newline

File: Makefile [hunk 2]
```diff
@@ -15,42 +15,38 @@
...
-
-docker/localkube:
-	docker run -w /go/src/k8s.io/minikube -v $(shell pwd):/go/src/k8s.io/minikube golang:1.6 make localkube-incremental
-
 .PHONY: test
-test: gopath
+test: .gopath
 	./test.sh
+
+pkg/minikube/cluster/localkubecontents.go: out/localkube
```
Comment (luxas) [inline Makefile]: Should this install `go-bindata` if not present?

File: README.md [hunk 3]
```diff
@@ -31,11 +31,16 @@ For more information about minikube, see the [proposal](https://github.com/kuber
 * Simplifying kubernetes production deployment experience. Kube-deploy is attempting to tackle this problem.
 * Supporting all possible deployment configurations of Kubernetes like various types of storage, networking, etc.
 
+## Build Requirements
+
+* A recent Go distribution (>1.5)
```
Comment (luxas) [inline README.md]: Please `go1.6` as supported to avoid problems

File: Makefile (hunk 2 above)
Comment (dlorenc) [inline Makefile]: Good idea. I stuck it in our .gopath so it won't interfere.

File: README.md (hunk 3 above)
Comment (dlorenc) [inline README.md]: Done. We can probably drop the: export GO15VENDOREXPERIMENT=1
from our Makefile eventually.

File: pkg/minikube/sshutil/sshutil.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+			return
+		}
+		header := fmt.Sprintf("C%s %d %s\n", perm, len(data), filename)
+		fmt.Fprint(w, header)
+		reader := bytes.NewReader(data)
+		io.Copy(w, reader)
+		fmt.Fprint(w, "\x00")
+	}()
+
+	scpcmd := fmt.Sprintf("sudo /usr/local/bin/scp -t %s", remotedir)
```
Comment (luxas) [inline pkg/minikube/sshutil/sshutil.go]: What about if `scp` is located in another location?
Maybe use a shell or query `which scp` first.
(This isn't so important...)

File: pkg/minikube/cluster/cluster.go [hunk 5]
```diff
@@ -154,6 +156,24 @@ func StartCluster(h sshAble, config KubernetesConfig) error {
 	return nil
 }
 
+func UpdateCluster(d drivers.Driver) error {
+	localkube, err := Asset("out/localkube")
```
Comment (luxas) [inline pkg/minikube/cluster/cluster.go]: Maybe move this to some constant? So we have only one place to update if we change the path in the future

File: pkg/minikube/cluster/commands.go [hunk 6]
```diff
@@ -1,17 +1,10 @@
...
-	sudo chmod a+x /usr/local/bin/localkube
-fi
+var startCommand = `
 # Run with nohup so it stays up. Redirect logs to useful places.
 PATH=/usr/local/sbin:$PATH nohup sudo /usr/local/bin/localkube start > /var/log/localkube.out 2> /var/log/localkube.err < /dev/null &
 `
 
-func getStartCommand(localkubeURL string) string {
-	return fmt.Sprintf(startCommand, localkubeURL)
+func getStartCommand() string {
```
Comment (luxas) [inline pkg/minikube/cluster/commands.go]: This function doesn't do much nowadays. Maybe remove.

**Comment by luxas** [comment]
This LGTM. 
Some nits here and there but we'll fix these in followups

Merging to move forward

**Comment by luxas** [comment]
Ah, wait. Travis returned false information again.
Fix this issue first:
```
# k8s.io/minikube/pkg/minikube/cluster
../../../k8s.io/minikube/pkg/minikube/cluster/cluster.go:160: undefined: Asset
FAIL	k8s.io/minikube/cmd/minikube/cmd [build failed]
```

**Comment by dlorenc** [comment]
> Ah, wait. Travis returned false information again.
> Fix this issue first:

Ah whoops, I think I got it this time.

File: Makefile [hunk 7]
```diff
@@ -15,42 +15,41 @@
...
-minikube: minikube-$(GOOS)-$(GOARCH)
+out/minikube: out/minikube-$(GOOS)-$(GOARCH)
 	cp $(BUILD_DIR)/minikube-$(GOOS)-$(GOARCH) $(BUILD_DIR)/minikube
 
-.PHONY: localkube
-localkube: localkube-$(GOOS)-$(GOARCH)
-	cp $(BUILD_DIR)/localkube-$(GOOS)-$(GOARCH) $(BUILD_DIR)/localkube
+out/localkube: .gopath
+ifeq ($(GOOS),linux)
+	CGO_ENABLED=1 go build -v -ldflags="-s" -o $(BUILD_DIR)/localkube ./cmd/localkube
```
Comment (luxas) [inline Makefile]: The `-v` here makes it very verbose all the time. 
Maybe add a `VERBOSE` flag to the makefile so we may control it?
I may do this later as well.

**Comment by luxas** [comment]
Seems like a `gofmt` failure

**Comment by dlorenc** [comment]
> Seems like a gofmt failure

Had to add the generated file and the .gopath directory to the whitelist. Let's see if we get farther this time.

**Comment by luxas** [comment]
LGTM, merging...

**Pull request merged**